contracts/Governance/Timelock.sol

```diff
@@ -1,123 +1,160 @@
+    event CancelledTransactionsBatch(bytes32 indexed txParamsHash);
+    event ApprovedTransactionsBatch(bytes32 indexed txParamsHash);
+    event ExecutedTransaction(bytes32 indexed txParamsHash, address indexed target, uint256 value, string signature, bytes data, uint256 eta);
+    event NewProposerSet(address indexed newProposer);
+    event NewDelaySet(uint256 indexed delay);
+
+    constructor(
+        address _proposer,
+        uint256 _minimumDelay,
+        uint256 _maximumDelay,
+        uint256 _gracePeriod,
+        uint256 _delay
+    ) {
+        require(_proposer != address(0), "Proposer address cannot be 0");
+        require(_minimumDelay <= _maximumDelay, "Minimum delay cannot be larger than maximum delay");
+
+        minimumDelay = _minimumDelay;
+        maximumDelay = _maximumDelay;
+        gracePeriod = _gracePeriod;
+
+        setDelay(_delay);
+
+        proposer = _proposer;
+    }
+
+    function setProposer(address _proposer) external onlyOwner {
+        require(_proposer != address(0), "Proposer address cannot be 0");
 
-    constructor(address admin_, uint256 delay_) {
-        require(admin_ != address(0), "Admin address cannot be 0");
-        require(delay_ >= MINIMUM_DELAY, "Timelock::constructor: Delay must exceed minimum delay.");
-        require(delay_ <= MAXIMUM_DELAY, "Timelock::setDelay: Delay must not exceed maximum delay.");
+        proposer = _proposer;
 
-        admin = admin_;
-        delay = delay_;
+        emit NewProposerSet(proposer);
     }
 
-    function setDelay(uint256 delay_) external {
-        require(msg.sender == address(this), "Timelock::setDelay: Call must come from Timelock.");
-        require(delay_ >= MINIMUM_DELAY, "Timelock::setDelay: Delay must exceed minimum delay.");
-        require(delay_ <= MAXIMUM_DELAY, "Timelock::setDelay: Delay must not exceed maximum delay.");
-        delay = delay_;
+    function setDelay(uint256 _delay) public onlyOwner {
+        require(_delay >= minimumDelay, "Timelock: Delay must exceed minimum delay.");
+        require(_delay <= maximumDelay, "Timelock: Delay must not exceed maximum delay.");
+
+        delay = _delay;
 
-        emit NewDelay(delay);
+        emit NewDelaySet(delay);
     }
 
-    function acceptAdmin() external {
-        require(msg.sender == pendingAdmin, "Timelock::acceptAdmin: Call must come from pendingAdmin.");
-        admin = msg.sender;
-        pendingAdmin = address(0);
+    function queueTransactionsBatch(Transaction[] memory transactions, uint256 eta) external onlyProposerOrOwner returns (bytes32) {
+        //todo ag hmm... do we really want it this way? we definitely want 0 delay on local deployment
+        require(eta >= block.timestamp + delay, "Timelock: Estimated execution time must satisfy delay.");
+        require(eta <= block.timestamp + delay + gracePeriod, "Timelock: Estimated execution time must satisfy delay and grace period.");
```

Comment:
I think the term `gracePeriod` and `delay` are a bit confusing here.
I think we can use different names that will make it clearer:
* `signingStartDelay`
* `executionPeriod` or `executionGracePeriod`

With that notion, also `eta` is confusing. ETA stands for - Estimated Time of Arrival, without reading the docs I cannot tell what it means in this context.
If the name was `executionStartTimestamp` it would have made it much clearer IMO.

WDYT?

Reply:
I changed `delay` to `executionDelay` I think it makes more sense sinde there's really no "signing". WDYT?
The rest I changes as you wanted

Reply:
Correct me if I'm wrong, but `delay` is not the `executionDelay`.
The delay is the delay for approving the transaction, not executing it.. The ETA is the time you can execute the transaction from. Isn't it?

Reply:
It's **not** approval delay. It's **execution** delay